app/src/main/java/com/android/favoritemakes/makeslist/MakesListScreen.kt

```diff
@@ -0,0 +1,122 @@
+package com.android.favoritemakes.makeslist
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.ImageLoader
+import coil.imageLoader
+import com.android.favoritemakes.R
+import com.android.favoritemakes.ui.theme.FavoriteMakesTheme
+
+@Composable
+fun MakesListScreen(
+    viewModel: MakesListViewModel = viewModel(),
+) {
+    val makes: List<MakeData> by viewModel.fetchMakes().collectAsState(initial = emptyList())
+    MakesList(makes = makes, viewModel.imageLoader) { makeId, isFavorite ->
+        viewModel.toggleFavoriteMake(makeId, isFavorite)
+    }
+}
+
+@Composable
+fun MakesList(
+    makes: List<MakeData>,
+    imageLoader: ImageLoader,
+    onFavoriteIconToggle: (Long, Boolean) -> Unit,
+) {
+    if (makes.isEmpty()) {
```

Comment:
just a small performance improvement - its a good pattern to use derivedStateOf  when composable depends on state calculated from list. This way it will be recomposed only when isEmpty changes not everytime when list changes

Reply:
Something like this:

```
val makes: List<MakeData> by viewModel.fetchMakes().collectAsState(initial = emptyList())
    val shouldShowEmptyState by remember(makes) {
        derivedStateOf { makes.isEmpty() }
    }
    MakesList(
        makes = makes,
        shouldShowEmptyState = shouldShowEmptyState,
        viewModel.imageLoader
    ) { makeId, isFavorite ->
        viewModel.toggleFavoriteMake(makeId, isFavorite)
    }
```

Reply:
Yep, you can also move it to MakesList, exactly where the `if` is. It doesnt matter if you pass it to composable function or use it inline so better to have it where its needed instead of two scopes